src/main/kotlin/com/google/androidstudiopoet/utils/DependencyImageGenerator.kt

```diff
@@ -0,0 +1,130 @@
+/*
+ *  Copyright 2017 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file
+ * except in compliance with the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the
+ * License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.google.androidstudiopoet.utils
+
+import com.google.androidstudiopoet.DependencyValidator
+import com.google.androidstudiopoet.models.ProjectBlueprint
+import java.awt.Color
+import java.awt.Graphics
+import java.awt.image.BufferedImage
+import java.io.File
+import javax.imageio.ImageIO
+
+class DependencyImageGenerator {
+    companion object{
+        const val CELL_SIZE = 3
+        const val LINE_WIDTH = 1
+        const val HEADER_SIZE = CELL_SIZE + 2 * LINE_WIDTH
+        const val GRID_SIZE = CELL_SIZE + LINE_WIDTH
+        val BACKGROUND_COLOR = Color(0xFFFFFF)
+        val GRID_COLOR = Color(0xAFAFAF)
+        val APP_COLOR = Color(0x5050FF)
+        val ANDROID_COLOR = Color(0xA4C639)
+        val JAVA_COLOR = Color(0xF7DB64)
+        val DEPENDENCY_COLOR = Color(0x00AF00)
+        val ERROR_COLOR = Color(0xAF0000)
+    }
+
+    /**
+     * generates an image that represents the dependency matrix of the project
+     */
+    fun generate(blueprint: ProjectBlueprint) {
+        val numModules = blueprint.moduleBlueprints.size + blueprint.androidModuleBlueprints.size
```

Comment:
please make `allModuleBlueprints` in `ProjectBlueprint` public and use it here.

Reply:
This part no longer exists, but made allModuleBlueprints public as it is required for the new changes